rust/azure_iot_operations_mqtt/src/aio/cloud_event.rs

```diff
@@ -0,0 +1,607 @@
+            "time" => Ok(CloudEventFields::Time),
+            _ => Err(()),
+        }
+    }
+}
+
+// ~~~~~~~~~~ Public CloudEvent struct and fns to convert to/from publishes ~~~~~~~~~~
+// ~~~~~~~~~~ Builder only used for creating a Cloud Event to send ~~~~~~~~~~
+
+/// Cloud Event struct.
+/// Note: if fields are modified after the [`CloudEvent`] has been built, there is no longer
+/// a guarantee that this is a valid cloud event.
+///
+/// Implements the cloud event spec 1.0 for the telemetry sender.
+/// See [CloudEvents Spec](https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md).
+#[derive(Builder, Clone, Debug)]
+#[builder(setter(into), build_fn(validate = "Self::validate"))]
+pub struct CloudEvent {
+    /// Identifies the context in which an event happened. Often this will include information such
+    /// as the type of the event source, the organization publishing the event or the process that
+    /// produced the event. The exact syntax and semantics behind the data encoded in the URI is
+    /// defined by the event producer.
+    pub source: String,
+    /// The version of the cloud events specification which the event uses. This enables the
+    /// interpretation of the context. Compliant event producers MUST use a value of 1.0 when
+    /// referring to this version of the specification.
+    #[builder(default = "DEFAULT_CLOUD_EVENT_SPEC_VERSION.to_string()")] // no default on recv
+    pub spec_version: String,
+    /// Contains a value describing the type of event related to the originating occurrence. Often
+    /// this attribute is used for routing, observability, policy enforcement, etc. The format of
+    /// this is producer defined and might include information such as the version of the type.
+    pub event_type: String,
+    /// Identifies the schema that data adheres to. Incompatible changes to the schema SHOULD be
+    /// reflected by a different URI.
+    #[builder(default = "None")]
+    pub data_schema: Option<String>,
+    /// Identifies the event. Producers MUST ensure that source + id is unique for each distinct
+    /// event. If a duplicate event is re-sent (e.g. due to a network error) it MAY have the same
+    /// id. Consumers MAY assume that Events with identical source and id are duplicates.
+    #[builder(default = "Uuid::new_v4().to_string()")] // no default on recv
+    pub id: String,
+    /// Timestamp of when the occurrence happened. If the time of the occurrence cannot be
+    /// determined then this attribute MAY be set to some other time (such as the current time) by
+    /// the cloud event producer, however all producers for the same source MUST be consistent in
+    /// this respect. In other words, either they all use the actual time of the occurrence or they
+    /// all use the same algorithm to determine the value used.
+    #[builder(default = "Some(DateTime::<Utc>::from(SystemTime::now()))")]
+    pub time: Option<DateTime<Utc>>,
```

Comment:
Why is it optional? The doc doesn't explain what a time of None would mean

Reply:
Not required in the Cloud Event spec https://github.com/cloudevents/spec/blob/main/cloudevents/spec.md#time